velox/substrait/VeloxToSubstraitExpr.cpp

```diff
@@ -49,6 +49,28 @@ void convertVectorValue(
   }
 }
 
+template <TypeKind flatKind>
+void flatVectorToListLiteral(
+    google::protobuf::Arena& arena,
+    const velox::VectorPtr& vector,
+    ::substrait::Expression_Literal_List* listLiteral) {
+  using T = typename TypeTraits<flatKind>::NativeType;
+  auto flatVector = vector->as<FlatVector<T>>();
+  VELOX_CHECK_NOT_NULL(flatVector, "flatVector can not be NULL");
+  const auto& type = flatVector->type();
+  for (int idx = 0; idx < flatVector->size(); idx++) {
+    ::substrait::Expression_Literal* childLiteral = listLiteral->add_values();
+    if (flatVector->isNullAt(idx)) {
+      // Process the null value.
+      childLiteral->MergeFrom(
+          exprConvertor_->toSubstraitNullLiteral(arena, type->kind()));
+    } else {
+      childLiteral->MergeFrom(exprConvertor_->toSubstraitNotNullLiteral(
```

Comment:
Do we need to convert a value of type T to variant? Can we convert T directly to Substrait's literal? This would avoid an extra type switch in toSubstraitNotNullLiteral.

Reply:
I thought about it. And I met some problem here. Because different typekind need use different set_xxx function, we still need TypeKind.

Reply:
I will use template function to implement toSubstraitNotNullLiteral.